webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala

```diff
@@ -0,0 +1,172 @@
+import zio.test._
+
+import dsp.valueobjects.Iri._
+import dsp.valueobjects.Project.ShortCode
+import dsp.valueobjects.Project._
+import dsp.valueobjects.V2
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectCreatePayloadADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectCreateRequestADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectGetRequestADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectGetResponseADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectIdentifierADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectOperationResponseADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectsGetRequestADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectsGetResponseADM
+import org.knora.webapi.messages.store.triplestoremessages.StringLiteralV2
+import org.knora.webapi.messages.util.KnoraSystemInstances
+import org.knora.webapi.responders.ActorToZioBridge
+import org.knora.webapi.responders.ActorToZioBridgeMock
+import org.knora.webapi.responders.admin.ProjectsService
+
+object ProjectsServiceLiveSpec extends ZIOSpecDefault {
+
+  private val expectNoInteraction = ActorToZioBridgeMock.empty
+
+  val layers = ZLayer.makeSome[ActorToZioBridge, ProjectsService](ProjectsService.live)
+
+  /**
+   * Creates a [[ProjectADM]] with empty content or optionally with a given ID.
+   */
+  private def getProjectADM(id: String = "") =
+    ProjectADM(
+      id = id,
+      shortname = "",
+      shortcode = "",
+      longname = None,
+      description = Seq(StringLiteralV2("")),
+      keywords = Seq.empty,
+      logo = None,
+      ontologies = Seq.empty,
+      status = true,
+      selfjoin = false
+    )
+
+  val getAllProjectsSpec = test("get all projects") {
+    val expectedResponse = ProjectsGetResponseADM(Seq(getProjectADM()))
+    val projectsService =
+      ZIO
+        .serviceWithZIO[ProjectsService](_.getProjectsADMRequest())
+        .provideSome[ActorToZioBridge](layers)
+    val bridge = ActorToZioBridgeMock.AskAppActor
+      .of[ProjectsGetResponseADM]
+      .apply(assertion = Assertion.equalTo(ProjectsGetRequestADM()), result = Expectation.value(expectedResponse))
+      .toLayer
+    for {
+      _ <- projectsService.provide(bridge)
+    } yield assertTrue(true)
+  }
+
+  val getProjectByIdSpec = suite("get project by identifier")(
+    test("get project by shortcode") {
+      val identifier = ProjectIdentifierADM.ShortcodeIdentifier
+        .fromString("0001")
+        .getOrElse(throw new Exception("invalid shortcode"))
+      val projectsService =
+        ZIO
+          .serviceWithZIO[ProjectsService](_.getSingleProjectADMRequest(identifier))
+          .provideSome[ActorToZioBridge](layers)
+      val bridge = ActorToZioBridgeMock.AskAppActor
```

Comment:
Would be nice to have speaking name for this mock layer.

Reply:
done